components/google-cloud/tests/experimental/custom_job/testdata/custom_job_pipeline.json

```diff
@@ -0,0 +1,156 @@
+{
+  "pipelineSpec": {
+    "components": {
+      "comp-producer": {
+        "executorLabel": "exec-producer",
+        "inputDefinitions": {
+          "parameters": {
+            "base_output_directory": {
+              "type": "STRING"
+            },
+            "encryption_spec_key_name": {
+              "type": "STRING"
+            },
+            "input_text": {
+              "type": "STRING"
+            },
+            "location": {
+              "type": "STRING"
+            },
+            "network": {
+              "type": "STRING"
+            },
+            "project": {
+              "type": "STRING"
+            },
+            "service_account": {
+              "type": "STRING"
+            },
+            "tensorboard": {
+              "type": "STRING"
+            }
+          }
+        },
+        "outputDefinitions": {
+          "parameters": {
+            "gcp_resources": {
+              "type": "STRING"
+            },
+            "output_value": {
+              "type": "STRING"
+            }
+          }
+        }
+      }
+    },
+    "deploymentSpec": {
+      "executors": {
+        "exec-producer": {
+          "container": {
+            "args": [
+              "--type",
+              "CustomJob",
+              "--payload",
+              "{\"display_name\": \"Producer\", \"job_spec\": {\"worker_pool_specs\": [{\"machine_spec\": {\"machine_type\": \"n1-standard-4\"}, \"replica_count\": 1, \"container_spec\": {\"image_uri\": \"google/cloud-sdk:latest\", \"command\": [\"sh\", \"-c\", \"set -e -x\\necho '$0, this is an output parameter' | gsutil cp - '$1'\\n\", \"{{$.inputs.parameters['input_text']}}\", \"{{$.outputs.parameters['output_value'].output_file}}\"]}, \"disk_spec\": {\"boot_disk_type\": \"pd-ssd\", \"boot_disk_size_gb\": 100}}], \"service_account\": \"{{$.inputs.parameters['service_account}']}}\", \"network\": \"{{$.inputs.parameters['network}']}}\", \"encryption_spec_key_name\": \"{{$.inputs.parameters['encryption_spec_key_name}']}}\", \"tensorboard\": \"{{$.inputs.parameters['tensorboard}']}}\", \"base_output_directory\": \"{{$.inputs.parameters['base_output_directory}']}}\"}}",
```

Comment:
`encryption_spec_key_name\": \"{{$.inputs.parameters['encryption_spec_key_name}']}}\`
doesn't form a correct custom job payload. It should map to encryption_spec.kms_key_name and you probably want to not set it if empty. otherwise it will fail with validation error complaining kms doesn't exist.

Reply:
if we want to allow the encryption key to be a pipeline parameter ( instead of hardcoded to the custom_job component payload we will need to support empty value for encryption_spec.kms_key_name. I will revert this to a string for now and add a tracker for supporting empty value on container side / serverless side.

Reply:
Alternatively we could only add the place holder when the user specifies a cmac key. and have that stored as the default.

Reply:
Done.